Source/WTF/wtf/SIMDHelpers.h

```diff
@@ -586,11 +586,31 @@ ALWAYS_INLINE simde_uint64x2_t greaterThanOrEqual(simde_uint64x2_t lhs, simde_ui
     return simde_vcgeq_u64(lhs, rhs);
 }
 
+template<size_t> struct SizedUnsignedTrait;
+template<>
+struct SizedUnsignedTrait<1> {
+    using Type = uint8_t;
+};
+template<>
+struct SizedUnsignedTrait<2> {
+    using Type = uint16_t;
+};
+template<>
+struct SizedUnsignedTrait<4> {
+    using Type = uint32_t;
+};
+template<>
+struct SizedUnsignedTrait<8> {
+    using Type = uint64_t;
+};
+template<typename T>
+using SizedUnsigned = SizedUnsignedTrait<sizeof(T)>::Type;
```

Comment:
I'm not sure I'd know what this name means without looking at the definition. Would something like `UnderlyingUnsignedInt` be easier to intuit (A la, `std::underlying_type` for enums)?

Reply:
Maybe. I would be happy to use your name but I won't use the abbreviation `Int` in it.

Reply:
On reflection, I don't think _underlying_ unsigned integer is a great way to name something mainly is used to map signed integer types to unsigned integer types. What this gives is an unsigned integer that's the same _size_ as the passed in type, not necessarily otherwise related to that type. That's a different concept than the enumeration where the underlying type truly is part of how the enumeration is implemented.

I'd love to give it a better name, but I don't think `UnderlyingUnsignedInteger` is better than, say, `SameSizeUnsignedInteger`. And I am not _sure_ a longer clearer name would make it better.